sphinx/util/__init__.py

```diff
@@ -281,7 +281,11 @@ def get_module_source(modname):
 
 
 def get_full_modname(modname, attribute):
-    # type: (str, unicode) -> unicode
+    # type: (str, unicode) -> Union[unicode, NoneType]
```

Comment:
I think you don't need to update here. `-> unicode` annotation expects the return value is a unicode or None.